packages/react-testing/src/root.tsx

```diff
@@ -257,15 +277,20 @@ export class Root<Props> implements Node<Props> {
     this.act(() => this.reactRoot!.unmount());
   }
 
-  destroy() {
+  async destroy() {
     const {element, mounted} = this;
 
     if (mounted) {
       this.unmount();
     }
-
     element.remove();
     connected.delete(this);
+    this.destroyed = true;
+    this.actingPromiseResolver();
+    // flush the micro task and then the macro task to wait until react commits all pending updates.
+    // flushing of the macro task is only required for React 17
+    await this.actingPromise;
+    await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
Is this last line really needed? It'd seem that the whole original promise chain for `act` should be effectively "canceled" at this point since all of the races would now resolve.

Reply:
React 17 was failing with the added tests and that seemed to have resolved it. I'm going to take a deeper look to see what's actually going on there.

Reply:
Didn't look like it was needed, I've removed it from here and moved it to the test instead to prevent React 17 from failing